Source/Core/Core/HW/DVD/DVDInterface.cpp

```diff
@@ -1512,26 +1512,33 @@ static void ScheduleReads(u64 offset, u32 length, const DiscIO::Partition& parti
     dvd_offset += DVD_ECC_BLOCK_SIZE;
   } while (length > 0);
 
-  // Update the buffer based on this read. Based on experimental testing,
-  // we will only reuse the old buffer while reading forward. Note that the
-  // buffer start we calculate here is not the actual start of the buffer -
-  // it is just the start of the portion we need to read.
+  // Update the buffer based on this read. Based on hardware testing, the blocks which are kept are
+  // the most recently accessed block, the block immediately before it, and all blocks after it.
+  // If the block immediately before the most recently accessed block is not kept, we get too long
+  // loading screens in Pitfall: The Lost Expedition (https://bugs.dolphin-emu.org/issues/1227).
```

Comment:
This link is wrong; it should be https://bugs.dolphin-emu.org/issues/12279 (it's missing the 9).

Also, "we get too long loading screens" sounds a bit awkward to me; maybe "loading takes longer than it should" would be better?